plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,501 @@
+			value := getFloat(fieldSet.Value)
+			// JSON does not support these special values
+			if math.IsInf(value, 1) {
+				s.Log.Debugf("metric %s returned positive infity, setting value to %f", key, math.MaxFloat64)
+				value = math.MaxFloat64
+			}
+			if math.IsInf(value, -1) {
+				s.Log.Debugf("metric %s returned negative infity, setting value to %f", key, -math.MaxFloat64)
+				value = -math.MaxFloat64
+			}
+			if math.IsNaN(value) {
+				s.Log.Debugf("metric %s returned as non a number, skipping", key)
+				continue
+			}
+
+			point := &OutputMetric{
+				Name:      metric.Name() + "." + key,
+				Tags:      tagList,
+				Timestamp: metric.Time().Unix(),
+				Value:     value,
+			}
+			points = append(points, point)
+		}
+	}
+
+	reqBody, err := s.encodeToJson(points)
+	if err != nil {
+		return err
+	}
+
+	return s.write(reqBody)
+}
+
+func (s *Sensu) write(reqBody []byte) error {
+	var reqBodyBuffer io.Reader = bytes.NewBuffer(reqBody)
+	method := http.MethodPost
+
+	if s.ContentEncoding == "gzip" {
+		rc, err := internal.CompressWithGzip(reqBodyBuffer)
+		if err != nil {
+			return err
+		}
+		defer rc.Close()
+		reqBodyBuffer = rc
+	}
+
+	req, err := http.NewRequest(method, s.EndpointUrl, reqBodyBuffer)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("User-Agent", internal.ProductToken())
+
+	req.Header.Set("Content-Type", defaultContentType)
+	if s.ContentEncoding == "gzip" {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
+
+	if s.ApiKey != nil {
+		req.Header.Set("Authorization", "Key "+*s.ApiKey)
+	}
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("when writing to [%s] received status code: %d", s.EndpointUrl, resp.StatusCode)
+	}
+
+	return nil
+}
+
+// Resolves the event write endpoint
+func (s *Sensu) setEndpointUrl() error {
+	var (
+		endpointUrl string
+		path_suffix string
+	)
+
+	if s.BackendApiUrl != nil {
+		endpointUrl = *s.BackendApiUrl
+		namespace := "default"
+		if s.Entity != nil && s.Entity.Namespace != nil {
+			namespace = *s.Entity.Namespace
+		}
+		path_suffix = "/api/core/v2/namespaces/" + namespace + "/events"
+	} else if s.AgentApiUrl != nil {
+		endpointUrl = *s.AgentApiUrl
+		path_suffix = "/events"
+	}
+
+	if len(endpointUrl) == 0 {
+		s.Log.Debugf("no backend or agent API URL provided, falling back to default agent API URL %s", defaultUrl)
+		endpointUrl = defaultUrl
+		path_suffix = "/events"
+	}
+
+	u, err := url.Parse(endpointUrl)
+	if err != nil {
+		return err
+	}
+
+	u.Path = path.Join(u.Path, path_suffix)
+	s.EndpointUrl = u.String()
+
+	return nil
+}
+
+func init() {
+	outputs.Add("sensu-go", func() telegraf.Output {
+		// Default configuration values
+
+		// make a string from the defaultUrl const
+		agentApiUrl := defaultUrl
+
+		return &Sensu{
+			AgentApiUrl: &agentApiUrl,
+			Timeout:     internal.Duration{Duration: defaultClientTimeout},
```

Comment:
can you switch to config.Duration? the usage is nicer and we're planning to deprecate internal.Duration.